packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp

```diff
@@ -237,7 +237,7 @@ namespace Teuchos
         // out of memory for storing callbacks).  We could throw an
         // exception here in that case, but I think it's better just
         // to let the minor memory leak happen.
-        (void) atexit (freeTableFormat);
```

Comment:
Removing `(void)` could cause errors for customers that have `-Werror` enabled, like Sierra.  Can we put in some conditional build-time logic to add `(void)` or not add `(void)` when needed?

Reply:
Sounds good. How about wrapping it around with the following?
```
static_cast<void>( atexit(freeTableFormat) );
```

This still builds fine with AMPI's macro trickery.

Reply:
@jbakosi, can you update the PR branch to use `static_cast<void>()`?  The automated PR testing should do a GCC 4.9.3 build and then we should be able to see if this generates a warning or not.

Reply:
The GCC 4.9.3 PR builds is running so we will see if it generates a warning.  I just hope this build GCC 4.9.3 build has high enough warning levels to catch what Sierra's BJAM build of Sierra APPs pulling in Trilinos headers would catch. It would be great if Sierra had a nightly build of Sierra building against Trilinos 'develop' so that we could catch problems like this early.

Reply:
Sure, see  ef98130.

I also doubt it will generate a warning, though. I use the following macro to wrap something that I want to a ignore compiler warning for, e.g., an unused variable:
```
#define IGNORE(expr) (static_cast<void>(expr))
```

Reply:
@jbakosi Thanks for the fix!  btw please don't introduce that macro into Trilinos without first namespacing it with the package name.  For example:

```c++
#ifndef TEUCHOS_IGNORE
#  define TEUCHOS_IGNORE( expr ) static_cast<void>(expr)
#endif
```

Reply:
Oh, no, I did not plan to. That is why, I just wrapped `atexit` around with the _contents_ of that macro.